proto/src/determined/api/v1/api.proto

```diff
@@ -509,6 +509,78 @@ service Determined {
       tags: "Internal"
     };
   }
+  // Get the absolute training length until the searcher needs the trial to
+  // report a validation.
+  rpc GetTrialSearcherTrainUntil(GetTrialSearcherTrainUntilRequest)
+      returns (GetTrialSearcherTrainUntilResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/trials/{trial_id}/searcher/train_until"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // Reports to the searcher that the trial has completed the current
+  // requested amount of training with the given searcher validation
+  // metric.
+  rpc ReportTrialSearcherValidation(ReportTrialSearcherValidationRequest)
+      returns (ReportTrialSearcherValidationResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/report_validation"
+      body: "searcher_metric"
+    };
+    option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
+      tags: "Internal"
+    };
+  }
+  // Reports to the searcher that the trial has completed the current
+  // requested amount of training with the given searcher validation
+  // metric.
+  rpc ReportTrialSearcherEarlyExit(ReportTrialSearcherEarlyExitRequest)
+      returns (ReportTrialSearcherEarlyExitResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/trials/{trial_id}/searcher/early_exit"
```

Comment:
early stopping is a more commonly used word in ML word if I am right

```suggestion
      post: "/api/v1/trials/{trial_id}/early_stop"
```

Reply:
yeah I agree, sgtm